discovery-infra/download_logs.py

```diff
@@ -50,31 +52,37 @@
 
 def main():
     args = handle_arguments()
+    report_name = f"junit_download_report_{args.cluster_id or str(uuid.uuid4())[:8]}"
```

Comment:
You shouldn't generate a UUID. Cluster ID should always be available either manually from the `args` or during the download.
The `report_name` should be generated as part of the main function itself

Reply:
what do you mean by `as part of the main function`?

Reply:
First, UUID shouldn't be generated. download_logs module would always work with some cluster_id. It's just not necessarily comes from an argument.
So I think that the definition of the `report_name` should be in a more internal scope.

BTW, this should work with multiple clusters. The `assisted-gather` step now generated 1 report while there are 3 clusters available. There should be 3 reports per cluster